pkg/admission/permissionclaims/mutating_permission_claims.go

```diff
@@ -93,29 +96,31 @@ func (m *mutatingPermissionClaims) Admit(ctx context.Context, a admission.Attrib
 	}
 
 	labels := u.GetLabels()
-	if labels == nil {
-		labels = make(map[string]string)
-	}
-
-	// Start by removing all claim labels
 	for k := range labels {
-		if strings.HasPrefix(k, apisv1alpha1.APIExportPermissionClaimLabelPrefix) {
+		if !strings.HasPrefix(k, apisv1alpha1.APIExportPermissionClaimLabelPrefix) {
+			continue
+		}
+		if _, expected := expectedLabels[k]; !expected {
 			delete(labels, k)
 		}
 	}
-
-	// Add back what should be there
 	for k, v := range expectedLabels {
+		if labels == nil {
+			labels = make(map[string]string)
+		}
 		labels[k] = v
 	}
-
 	u.SetLabels(labels)
 
 	return nil
 
 }
 
 func (m *mutatingPermissionClaims) Validate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) error {
+	if a.GetSubresource() != "" {
```

Comment:
Is it always the case that we don't want to validate the labels the subresource? 

I am unaware of a subresource that will let you change metadata, but I thought I would ask.

Reply:
as admission is best effort in our case this should be safe, even if some day some other subresource is added that allows it (I doubt it).